### DISCUSSION TITLE
Why are request cookie values typed as `string | string[]` instead of just `string`?

### DISCUSSION THREAD

**Discussion by ezzatron** [mswjs/msw#1854] [Q&A] [answered]
To clarify, this is an MSW 2 question, and I'm specifically talking about the `cookies` property of the handler `info` object when using `http` (probably `graphql` too), like so:

```ts
import { http } from "msw";

export const handlers = [
  http.get("/", ({ cookies }) => {
    // `cookies` has type `Record<string,string|string[]>`
  })
]
```

I could be misunderstanding something, but I can't think of a circumstance where cookie values make sense to represent with an array? It's extra weird, since there are other places in the code where MSW represents cookies with `Record<string,string>` 🤔

**Comment by kettanaito** [comment, marked as answer]
Hey, @ezzatron. Thanks for raising this. I believe this is a bug in types. While response cookies may have multi-value cookies, like `Set-Cookie`, request cookies are always a flat `[string]: string` map.

Would you like to open a pull request to fix this?

**Reply by ezzatron** [reply]
Sure! I'll take a look.

**Reply by ezzatron** [reply]
Submitted: https://github.com/mswjs/msw/pull/1857
